--- plugin.hpp-	2023-07-19 22:34:23.401282200 +0200
+++ plugin.hpp	2023-12-28 00:52:55.612329400 +0100
@@ -1,9 +1,11 @@
 // validator: no-bom
 #ifndef PLUGIN_HPP_3FC978E9_63BE_4FC2_8F96_8188B0AF8363
 #define PLUGIN_HPP_3FC978E9_63BE_4FC2_8F96_8188B0AF8363
+#if defined(NO_PYTHON_PLUGIN)
 #pragma once
+#endif
 
 /*
 plugin.hpp
 
 Plugin API for Far Manager 3.0.6161.0
@@ -47,23 +49,28 @@
 #define FARMANAGERVERSION_BUILD 6161
 #define FARMANAGERVERSION_STAGE VS_PRIVATE
 
 #ifndef RC_INVOKED
 
+#if defined(NO_PYTHON_PLUGIN)
 #include <stdint.h>
 #include <windows.h>
+#endif
 
 #undef DefDlgProc
 
 #ifdef __cpp_inline_variables
 #define FAR_INLINE_CONSTANT inline constexpr
 #else
 #define FAR_INLINE_CONSTANT static const
 #endif
 
+#if defined(NO_PYTHON_PLUGIN)
 typedef GUID UUID;
-
+#else
+#define UUID GUID
+#endif
 
 #define CP_UNICODE    ((uintptr_t)1200)
 #define CP_REVERSEBOM ((uintptr_t)1201)
 #define CP_DEFAULT    ((uintptr_t)-1)
 #define CP_REDETECT   ((uintptr_t)-2)
@@ -266,10 +273,11 @@
 	DI_LISTBOX                      = 11,
 
 	DI_USERCONTROL                  =255,
 };
 
+#if defined(NO_PYTHON_PLUGIN)
 /*
    Check diagol element type has inputstring?
    (DI_EDIT, DI_FIXEDIT, DI_PSWEDIT, etc)
 */
 static __inline BOOL IsEdit(enum FARDIALOGITEMTYPES Type)
@@ -283,10 +291,11 @@
 			return TRUE;
 		default:
 			return FALSE;
 	}
 }
+#endif
 
 typedef unsigned long long FARDIALOGITEMFLAGS;
 FAR_INLINE_CONSTANT FARDIALOGITEMFLAGS
 	DIF_BOXCOLOR              = 0x0000000000000200ULL,
 	DIF_GROUP                 = 0x0000000000000400ULL,
@@ -2101,14 +2110,20 @@
 	FCT_ANY=0,
 	FCT_STREAM=1,
 	FCT_COLUMN=2
 };
 
+#if defined(NO_PYTHON_PLUGIN)
 // <C&C++>
 typedef int (WINAPIV *FARSTDSPRINTF)(wchar_t *Buffer,const wchar_t *Format,...);
 typedef int (WINAPIV *FARSTDSNPRINTF)(wchar_t *Buffer,size_t Sizebuf,const wchar_t *Format,...);
 typedef int (WINAPIV *FARSTDSSCANF)(const wchar_t *Buffer, const wchar_t *Format,...);
+#else
+typedef int (WINAPIV *FARSTDSPRINTF)(wchar_t *Buffer,const wchar_t *Format);
+typedef int (WINAPIV *FARSTDSNPRINTF)(wchar_t *Buffer,size_t Sizebuf,const wchar_t *Format);
+typedef int (WINAPIV *FARSTDSSCANF)(const wchar_t *Buffer, const wchar_t *Format);
+#endif
 // </C&C++>
 typedef void (WINAPI *FARSTDQSORT)(void *base, size_t nelem, size_t width, int (WINAPI *fcmp)(const void *, const void *,void *userparam),void *userparam);
 typedef void   *(WINAPI *FARSTDBSEARCH)(const void *key, const void *base, size_t nelem, size_t width, int (WINAPI *fcmp)(const void *, const void *,void *userparam),void *userparam);
 typedef size_t (WINAPI *FARSTDGETFILEOWNER)(const wchar_t *Computer,const wchar_t *Name,wchar_t *Owner,size_t Size);
 typedef size_t (WINAPI *FARSTDGETNUMBEROFLINKS)(const wchar_t *Name);
@@ -2417,10 +2432,11 @@
 	DWORD Revision;
 	DWORD Build;
 	enum VERSION_STAGE Stage;
 };
 
+#if defined(NO_PYTHON_PLUGIN)
 static __inline BOOL CheckVersion(const struct VersionInfo* Current, const struct VersionInfo* Required)
 {
 	return
 		(Current->Major > Required->Major) ||
 		(Current->Major == Required->Major && Current->Minor > Required->Minor) ||
@@ -2431,10 +2447,11 @@
 static __inline struct VersionInfo MAKEFARVERSION(DWORD Major, DWORD Minor, DWORD Revision, DWORD Build, enum VERSION_STAGE Stage)
 {
 	const struct VersionInfo Info = {Major, Minor, Revision, Build, Stage};
 	return Info;
 }
+#endif
 
 #define FARMANAGERVERSION MAKEFARVERSION(FARMANAGERVERSION_MAJOR,FARMANAGERVERSION_MINOR, FARMANAGERVERSION_REVISION, FARMANAGERVERSION_BUILD, FARMANAGERVERSION_STAGE)
 
 struct GlobalInfo
 {
@@ -2931,10 +2948,11 @@
 };
 
 FAR_INLINE_CONSTANT UUID FarGuid =
 {0x00000000, 0x0000, 0x0000, {0x00,0x00, 0x00,0x00,0x00,0x00,0x00,0x00}};
 
+#if defined(NO_PYTHON_PLUGIN)
 #ifdef __cplusplus
 extern "C"
 {
 #endif
 // Exported Functions
@@ -2972,9 +2990,10 @@
 	void     WINAPI FreeContentDataW(const struct GetContentDataInfo *Info);
 
 #ifdef __cplusplus
 }
 #endif
+#endif
 
 #endif /* RC_INVOKED */
 
 #endif // PLUGIN_HPP_3FC978E9_63BE_4FC2_8F96_8188B0AF8363
